### ISSUE TITLE
mastering-pytorch-from-linear-regression-to-computer-vision - improvements

### ISSUE THREAD

**Issue by JOTOR** [imteekay/iamtk.co#267]
Link: content/mastering-pytorch-from-linear-regression-to-computer-vision/en/index.mdx

First of all, thank you so much for sharing this helpful and insightful content.  However, I was wondering if the following improvements can be performed:

- Add the device = torch.device() so the example code can run in both 'CPU' and 'GPU'
- Update the code to work with 'device' so the items and labels can be shared or sent to the 'device'.
- On the following lines:
- `second_row_third_column = tensor[1, 2]
last_row = t[-1]
second_row_third_column # tensor(7)
last_row                # tensor([9, 10, 11, 12])`

I think `t` should be `tensor`.

IMHO, all these changes can help to replicate your code to work with a GPU (train the image classifiers). 

Thanks.

**Comment by imteekay** [comment]
Thanks for the feedback @JOTOR! 

For the tensor name, I have just fixed it https://github.com/imteekay/iamtk.co/commit/27cf1d7cb9c58aa75305769fefc3efc335985fed

For the cpu/gpu device, it can definitely be added to the training process. I would need to implement that on my notebook and test it out before updating the content, just to make sure it works properly. In the meantime, I can add a note to the blogpost saying that anyone can try gpu out if they have access to it.

**Comment by imteekay** [comment]
Just added a small note while I don't have a full notebook using cuda. https://github.com/imteekay/iamtk.co/commit/22748edc7200011de6d6fb3fee7abae189640b51
